src/store/modules/endpoints.js

```diff
@@ -1,129 +1,122 @@
-					: `/index.php/apps/openconnector/api/endpoints/${this.endpointItem.id}`
-				const method = isNewEndpoint ? 'POST' : 'PUT'
-
-				// Create a copy of the endpoint item and remove empty properties
-				const endpointToSave = { ...this.endpointItem }
-				Object.keys(endpointToSave).forEach(key => {
-					if (endpointToSave[key] === '' || (Array.isArray(endpointToSave[key]) && !endpointToSave[key].length)) {
-						delete endpointToSave[key]
-					}
-				})
-
-				return fetch(
-					endpoint,
-					{
-						method,
-						headers: {
-							'Content-Type': 'application/json',
-						},
-						body: JSON.stringify(endpointToSave),
+export const useEndpointStore = defineStore('endpoint', {
+	state: () => ({
+		endpointItem: false,
+		endpointList: [],
+	}),
+	actions: {
+		setEndpointItem(endpointItem) {
+			this.endpointItem = endpointItem && new Endpoint(endpointItem)
+			console.log('Active endpoint item set to ' + endpointItem)
+		},
+		setEndpointList(endpointList) {
+			this.endpointList = endpointList.map(
+				(endpointItem) => new Endpoint(endpointItem),
+			)
+			console.log('Endpoint list set to ' + endpointList.length + ' items')
+		},
+		/* istanbul ignore next */ // ignore this for Jest until moved into a service
+		async refreshEndpointList(search = null) {
+			// @todo this might belong in a service?
+			let endpoint = '/index.php/apps/openconnector/api/endpoints'
+			if (search !== null && search !== '') {
+				endpoint = endpoint + '?_search=' + search
+			}
+
+			const response = await fetch(endpoint, {
+				method: 'GET',
+			})
+
+			const data = (await response.json()).results
+			const entities = data.map(endpointItem => new Endpoint(endpointItem))
+
+			this.setEndpointList(data)
+
+			return { response, data, entities }
+		},
+		// New function to get a single endpoint
+		async getEndpoint(id) {
+			const endpoint = `/index.php/apps/openconnector/api/endpoints/${id}`
+
+			const response = await fetch(endpoint, {
+				method: 'GET',
+			})
+
+			const data = await response.json()
+			const entity = new Endpoint(data)
+
+			this.setEndpointItem(data)
+
+			return { response, data, entity }
+		},
+		// Delete an endpoint
+		async deleteEndpoint(endpointItem) {
+			if (!endpointItem) {
+				throw new MissingParameterError('endpointItem')
+			}
+
+			console.log('Deleting endpoint...')
+
+			const endpoint = `/index.php/apps/openconnector/api/endpoints/${endpointItem.id}`
+
+			const response = await fetch(endpoint, {
+				method: 'DELETE',
+			})
+
+			response.ok && this.setEndpointItem(null)
+			this.refreshEndpointList()
+
+			return { response }
+		},
+		// Create or save an endpoint from store
+		async saveEndpoint(endpointItem) {
+			if (!endpointItem) {
+				throw new MissingParameterError('endpointItem')
+			}
+
+			// convert to an entity
+			endpointItem = new Endpoint(endpointItem)
+
+			// verify data with Zod
+			const validationResult = endpointItem.validate()
```

Comment:
When I update an endpoint I get multiple errors.

1. Id is number not string.
2. Created is not a valid date time
3. Updated is not a valid date time

These errors should not come back since id should be set as a string (which is backend can be fixed in frontend)
And Created and Updated is set in the backend and should not be checked or validated when updating